### DISCUSSION TITLE
Distinguish Constants and Enum Members

### DISCUSSION THREAD

**Discussion by nadeeshaan** [ballerina-platform/ballerina-lang#28967] [Ideas]
In the current compiler implementation, the Enum Members are being converted to Constants. Hence, when we access the particular symbols in the semantic API as well as third party compiler API consumers such as the language server, it is necessary to distinguish the constants and the enum members. In the language server we need to assign a completion item kind for the enum members and this cannot achieved due to $subject.

@hasithaa @pubudu91 @rasika @IMS94 thoughts?

**Comment by IMS94** [comment]
@nadeeshaan +1. By a completion item kind, you mean we just need to show the completion item kind as `Enum` just like `Function`, `Struct`, etc? Or we need this to provide completions for `EnumName.<cursor>` scenarios?

**Reply by nadeeshaan** [reply]
@IMS94 yes, as `Enum` and `Function` even the enum members should be identifiable. Currently all the enum members are identified as Constants. This has a major impact on the completion Items sorting

**Comment by pubudu91** [comment]
+1 for this. Is there an issue for this? If not, can you create one?

**Reply by nadeeshaan** [reply]
@pubudu91 please refer this issue
https://github.com/ballerina-platform/ballerina-lang/issues/28969

**Comment by pubudu91** [comment]
For this, will it be enough to be able to identify a constant symbol as being an enum member? For example, was thinking along the lines of introducing a new symbol kind, `ENUM_MEMBER`, and for enum member constant symbols, return this when `kind()` is called.

**Reply by nadeeshaan** [reply]
I think that will be okay for now, because if we are to introduce a separate Symbol for enum fields, then we will have to change the API of the `BallerinaEnumSymbol` as well. Hence I'm +1 for the kind change. Shall we proceed with it then?

**Reply by pubudu91** [reply]
Yeah exactly. We'll have to deprecate `members()` in `EnumSymbol` and the newly introduced enum member symbol won't be adding anything new except change the symbol kind.

**Reply by nadeeshaan** [reply]
@pubudu91 yes. Shall we proceed with the `SymbolKind` approach then?

**Reply by pubudu91** [reply]
+1. Will proceed with that. @hasithaa please note.
